base/heartbeat.go

```diff
@@ -198,66 +223,28 @@ func (h *couchbaseHeartBeater) checkStaleHeartbeats(staleThresholdMs int, handle
 
 			// doc not found, which means the heartbeat doc expired.
 			// call back the handler.
-			handler.StaleHeartBeatDetected(heartbeatDoc.NodeUUID)
+			handler.StaleHeartBeatDetected(heartbeatDocID)
 
 			// delete the heartbeat doc itself so we don't have unwanted
 			// repeated callbacks to the stale heartbeat handler
-			docId := h.heartbeatDocId(heartbeatDoc.NodeUUID)
-			if err := h.bucket.Delete(docId); err != nil {
-				Infof(KeyImport, "Failed to delete heartbeat doc: %v err: %v", docId, err)
-			}
-
+			h.heartbeatHandler.RemoveNode(heartbeatDocID)
```

Comment:
Should we move the info logging from `viewBackedNodeListHandler.RemoveNode` up into here, to cover the `documentBackedNodeListHandler` implementation too?

```suggestion
			if err := h.heartbeatHandler.RemoveNode(heartbeatDocID); err != nil {
				Infof(KeyImport, "Failed to remove node %q: %v", heartbeatDocID, err)
			}
```

Reply:
Agreed